Client/src/main/java/configuration/ClientConfiguration.java

```diff
@@ -0,0 +1,138 @@
+
+    @Bean
+    public Publisher boardPublisher() {
+        return new Publisher();
+    }
+
+
+    @Bean
+    public ShipManager shipManager() {
+        return new ShipManager();
+    }
+
+    @Bean
+    public BelowPanelsListenerFactory belowPanelsListenerFactory() {
+        return new BelowPanelsListenerFactory();
+    }
+
+    @Bean
+    public BelowPanelsFactory belowPanelsFactory() {
+        return new BelowPanelsFactory();
+    }
+
+    @Bean
+    public UserBelowPanel userBelowPanel() {
+        return new UserBelowPanel();
+    }
+
+    @Bean
+    public HttpInvokerProxyFactoryBean shipGeneratorHttpInvokerProxyFactoryBean() {
+        HttpInvokerProxyFactoryBean httpInvoker = new HttpInvokerProxyFactoryBean();
+        String serviceURL = "http://" + Config.HOST_NAME + ":" + Config.SERVER_PORT + Config.SHIP_GENERATOR_SERVICE;
+        httpInvoker.setServiceInterface(ShipGeneratorService.class);
+        httpInvoker.setServiceUrl(serviceURL);
+        return httpInvoker;
+    }
+
+    @Bean
+    public HttpInvokerProxyFactoryBean boardMessageHttpInvokerProxyFactoryBean() {
+        HttpInvokerProxyFactoryBean httpInvoker = new HttpInvokerProxyFactoryBean();
+        String serviceURL = "http://" + Config.HOST_NAME + ":" + Config.SERVER_PORT + Config.BOARD_MESSAGE_SERVICE;
+        httpInvoker.setServiceInterface(BoardsMessageService.class);
+        httpInvoker.setServiceUrl(serviceURL);
+        return httpInvoker;
+    }
+
+    @Bean
+    public HttpInvokerProxyFactoryBean playerRegistrationHttpInvokerProxyFactoryBean() {
+        HttpInvokerProxyFactoryBean httpInvoker = new HttpInvokerProxyFactoryBean();
+        String serviceURL = "http://" + Config.HOST_NAME + ":" + Config.SERVER_PORT + Config.PLAYER_REGISTRATION_SERVICE;
+        httpInvoker.setServiceInterface(PlayerRegistrationService.class);
+        httpInvoker.setServiceUrl(serviceURL);
+        return httpInvoker;
+    }
+
+    @Bean
+    public HttpInvokerProxyFactoryBean gameInitHttpInvokerProxyFactoryBean() {
+        HttpInvokerProxyFactoryBean httpInvoker = new HttpInvokerProxyFactoryBean();
+        String serviceURL = "http://" + Config.HOST_NAME + ":" + Config.SERVER_PORT + Config.GAME_INIT_SERVICE;
+        httpInvoker.setServiceInterface(GameInitService.class);
+        httpInvoker.setServiceUrl(serviceURL);
+        return httpInvoker;
+    }
+
+    @Bean
+    public HttpInvokerProxyFactoryBean shootServiceHttpInvokerProxyFactoryBean() {
+        HttpInvokerProxyFactoryBean httpInvoker = new HttpInvokerProxyFactoryBean();
+        String serviceURL = "http://" + Config.HOST_NAME + ":" + Config.SERVER_PORT + Config.SHOOT_SERVICE;
+        httpInvoker.setServiceInterface(ShootService.class);
+        httpInvoker.setServiceUrl(serviceURL);
+        return httpInvoker;
+    }
+
+    @Bean
+    public HttpInvokerProxyFactoryBean endGameServiceHttpInvokerProxyFactoryBean() {
+        HttpInvokerProxyFactoryBean httpInvoker = new HttpInvokerProxyFactoryBean();
```

Comment:
Why not DRY? All the proxy factory bean invokers here need one method with two params, right?